apis/bigquery/v1beta1/dataset_types.go

```diff
@@ -136,11 +135,17 @@ type BigQueryDatasetStatus struct {
 	// Output only. A URL that can be used to access the resource again. You can
 	//  use this URL in Get or Update requests to the resource.
 	SelfLink *string `json:"selfLink,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *BigQueryDatasetObservedState `json:"observedState,omitempty"`
 }
 
-// BigQueryDatasetSpec defines the desired state of BigQueryDataset
+// BigQueryDatasetObservedState defines the desired state of BigQueryDataset
 // +kcc:proto=google.cloud.bigquery.v2.dataset
 type BigQueryDatasetObservedState struct {
+
+	// Optional. If the location is not specified in the spec, the GCP server defaults to a location and will be captured here.
+	Location *string `json:"location,omitempty"`
```

Comment:
Not sure we want to add location to observedState at this point. The default controller is still Terraform-based, and the direct controller doesn't use this field.

Reply:
Yes the default controller is still TF. I added this field with the intention to use it in our update call. The idea is that if `spec.Location` is not specified upon creation, I want to use the location value in the `observedState` during an UPDATE call and compared that with actual. I have updated the code logic.